### PR TITLE
[SYCL][InvokeSIMD][Doc] Remove revision history table

### PR DESCRIPTION
**Pull request by sarnex**
The invoke SIMD spec revision table was out of date, and we don't use it anymore in the latest spec template, so just remove it.

### CODE REVIEW

**Comment by Pennycook** [comment]
@sarnex: We don't actually use these revision tables anymore (see the template [here](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/template.asciidoc)).  If you wanted to, you could just remove it entirely.

**Review by Pennycook — APPROVED** [review]
Approving regardless of whether the table is updated or removed.  Somebody should make another pass to make sure the invoke_simd documentation follows the new template.

**Comment by sarnex** [comment]
@Pennycook Ah ok thanks, I'll just remove it

**Comment by bader** [comment]
> We don't use this table anymore, see below conversation

@sarnex, could you summarize the conversation in the description, please? PR description is used "as is" for git commit log message. "see below conversation" doesn't look useful as a git log message.

**Comment by sarnex** [comment]
@bader Fixed
